repl/src/main/scala/org/apache/spark/repl/ExecutorClassLoader.scala

```diff
@@ -71,27 +72,64 @@ class ExecutorClassLoader(conf: SparkConf, classUri: String, parent: ClassLoader
     }
   }
 
+  private def getClassFileInputStreamFromHttpServer(pathInDirectory: String): InputStream = {
+    val url: URL = if (SparkEnv.get.securityManager.isAuthenticationEnabled()) {
+      val uri = new URI(classUri + "/" + urlEncode(pathInDirectory))
+      Utils.constructURIForAuthentication(uri, SparkEnv.get.securityManager).toURL
+    } else {
+      new URL(classUri + "/" + urlEncode(pathInDirectory))
+    }
+    val connection = url.openConnection().asInstanceOf[HttpURLConnection]
+    if (connection.asInstanceOf[HttpURLConnection].getResponseCode != 200) {
```

Comment:
Yep, even this is still subtly-unsafe due to HttpUrlConnection's weird KeepAlive behavior:

https://plumbr.eu/blog/12-year-old-bug-in-jdk-still-out-there-leaking-memory
https://news.ycombinator.com/item?id=4931911